### PR TITLE
Support transformers v5 rope_parameters in model builder

### PR DESCRIPTION
**Pull request by tianleiwu**
## Summary

Make the ONNX model builder read RoPE settings from the config in a way that works across transformers versions. Transformers v5 standardizes RoPE metadata under `rope_parameters` (keeping `rope_scaling` only as a backward‑compatible alias), so the builder now reads whichever field the installed transformers version provides instead of relying solely on `rope_scaling`.

## Motivation

For models that carry their RoPE settings inside the config dict (e.g. GPT‑OSS YaRN, Llama‑3.1, Phi‑3 LongRoPE, Qwen MRoPE), transformers v5 moved those keys — including `original_max_position_embeddings`, `rope_theta`, `factor`, `beta_fast`/`beta_slow`, `mrope_section` — from `rope_scaling` into `rope_parameters`. The builder read them only via `config.rope_scaling`. If a config exposes only `rope_parameters` (or a future transformers drops the `rope_scaling` alias), `hasattr(config, "rope_scaling")` fails and the builder silently falls back to defaults:

- `original_context_length` falls back to the **extended** context length instead of the pretraining length. For GPT‑OSS YaRN this shifts the correction range and leaves rotary dimensions unscaled, degrading long‑context accuracy.
- `make_rope_init` is skipped, so `mscale`, YaRN `factor`, and `rope_theta` are never applied.

## Key Changes

Added a shared helper `Model.get_rope_parameters(config)` that prefers `rope_parameters` and falls back to `rope_scaling`, and routed every RoPE‑config read through it.

| File | Change |
|---|---|
| `base.py` | New `get_rope_parameters()` staticmethod; use it for `original_context_length`, `rope_theta`, the `make_rope_init` trigger, and the YaRN/linear/MRoPE branches in `make_rope_init`. |
| `llama.py` | `make_rope_init` reads `low_freq_factor`/`high_freq_factor` via the helper. |
| `phi.py` | `make_rope_init` (Phi‑3 LongRoPE short/long factors) via the helper. |
| `qwen.py` | Qwen2.5‑VL rope‑type check and Qwen3.5 `rope_theta`/`partial_rotary_factor` extraction via the helper. |
| `hunyuan.py` | Reuse the shared helper instead of an inline `getattr` fallback. |

No behavior change on older transformers (where `rope_scaling` is present); the helper resolves to the same dict.

## Testing Notes

- `python -m py_compile` and `lintrunner -a` pass clean on all five files.
- Verified with transformers 5.10.2 that the GPT‑OSS config resolves `original_context_length = 4096` (not `131072`).
- Verified the helper resolves correctly across three shapes: real config (`rope_parameters` + `rope_scaling` alias), `rope_parameters`‑only (future), and `rope_scaling`‑only (older transformers) — all yield the expected values.

### CODE REVIEW

File: src/python/py/models/builders/base.py [hunk 1]
```diff
@@ -391,18 +392,30 @@ def make_outputs_init(self):
...
+    def get_rope_parameters(config):
+        """Return the RoPE parameters mapping from the model config.
+
+        Transformers v5 standardizes RoPE settings under `rope_parameters`.
+        Older versions (and the v5 backward-compatible alias) expose the same
+        data under `rope_scaling`. Prefer `rope_parameters` and fall back to
+        `rope_scaling` so the builder works across transformers versions.
+        """
+        return getattr(config, "rope_parameters", None) or getattr(config, "rope_scaling", None)
+
```
[the remark this answers is not in the document]
Comment (tianleiwu) [reply src/python/py/models/builders/base.py]: Fixed in 5862679712. `get_rope_parameters` now returns `rope_parameters` whenever the attribute is present (explicit `is not None` check) instead of using `or`, so a present-but-empty `{}` is treated as authoritative and no longer silently falls back to `rope_scaling`. Added `test_get_rope_parameters_empty_rope_parameters_not_overridden` to lock this in.

File: src/python/py/models/builders/base.py [hunk 2]
```diff
@@ -391,18 +392,30 @@ def make_outputs_init(self):
...
     def make_rope_init(self, config):
-        if "beta_fast" in config.rope_scaling:
+        rope_params = self.get_rope_parameters(config)
+        if "beta_fast" in rope_params:
             # For models that use YARN (e.g. OpenAI OS-minier, Ministral3)
```
[the remark this answers is not in the document]
Comment (tianleiwu) [reply src/python/py/models/builders/base.py:413]: Fixed in 5862679712. `make_rope_init` now guards with `if not isinstance(rope_params, Mapping): return` before any `in` check, so a missing/non-mapping value is a safe no-op instead of raising `"beta_fast" in None`. The same guard was added to the `llama.py` and `phi.py` overrides. Covered by `test_make_rope_init_no_op_without_rope_params`.

File: src/python/py/models/builders/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tianleiwu) [reply src/python/py/models/builders/base.py]: Added in 5862679712. New regression tests drive the base builder RoPE path with a config that provides **only** `rope_parameters` (no `rope_scaling` attribute): `test_rope_parameters_only_config_matches_rope_scaling` (cos/sin parity vs. the HF reference and vs. the legacy `rope_scaling` shape) and `test_rope_parameters_only_resolves_original_context_length` (guards against the 131072 regression). Also added focused unit tests for `get_rope_parameters` preference/fallback/empty/None.

File: src/python/py/models/builders/qwen.py [hunk 3]
```diff
@@ -61,8 +61,9 @@ def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
 
         # Check rope type since huggingface model supports yarn but that is not recommended as mentioned in model card. Example:
         #    "rope_scaling": {"type": "mrope", "mrope_section": [16, 24,24]}
-        if config.rope_scaling and "type" in config.rope_scaling:
-            assert config.rope_scaling["type"] in ["mrope", "default"]
+        rope_params = self.get_rope_parameters(config)
+        if rope_params and "type" in rope_params:
+            assert rope_params["type"] in ["mrope", "default"]
```
[the remark this answers is not in the document]
Comment (tianleiwu) [reply src/python/py/models/builders/qwen.py:66]: Fixed in 5862679712. Both MRoPE "sections not found" messages (Qwen2.5-VL and Qwen3.5) now read `MRoPE sections not found in text_config rope_parameters/rope_scaling mrope_section`, so they no longer point only at the legacy `rope_scaling` path.

File: src/python/py/models/builders/base.py [hunk 4]
```diff
@@ -391,18 +392,35 @@ def make_outputs_init(self):
         if self.exclude_lm_head:
             del self.output_names["logits"]
 
+    @staticmethod
```
Comment (kunal-vaishnavi) [inline src/python/py/models/builders/base.py]: Can we remove the static method decorator on this method?

**Review by kunal-vaishnavi — APPROVED** [review]
